src/InvocationModel.ts

```diff
@@ -151,8 +151,8 @@ export class InvocationModel implements coreTypes.InvocationModel {
         this.#log(level, 'system', ...args);
     }
 
-    #userLog(level: RpcLogLevel, ...args: any[]): void {
-        if (this.#isDone && this.#coreCtx.state !== 'postInvocationHooks') {
+    #userLog(level: RpcLogLevel, context: Context, ...args: any[]): void {
+        if (this.#isDone && this.#coreCtx.state !== 'postInvocationHooks' && !context.suppressBadPatternError) {
```

Comment:
I was thinking of this as more of a "let me do what I want, I know what I'm doing" kind of flag. If a user has some use case where they think it's valid to call `context.done()` in an async function, they may also, for the same reason, want to log after calling `context.done()`

Reply:
They may also think that, but they should have to set another flag to suppress it. The user should have to explicitly choose to ignore each and every warning - we have those warnings for a good reason

Reply:
Sure. Should I also add that second flag, or wait to see if users ask for it?

Reply:
Wait to see if people ask for it. Even if they ask, we will need a compelling reason (in Durable's case, the compelling reason is because they can't fix for backwards compat reasons)

Reply:
Fair :)

Reply:
Removed this